sbt

```diff
@@ -707,6 +707,9 @@ loadConfigFile() {
 }
 
 loadPropFile() {
+  # trim key and value so as to be more forgiving with spaces around the '=':
+  k=$(echo $k |sed -e 's/^\s*(.+)\s*$/\\1/g')
+  v=$(echo $v |sed -e 's/^\s*(.+)\s*$/\\1/g')
```

Comment:
Just wondering if it is possible to solve this with builtin bash functionality to avoid forking a new process. E.g. something like to replace all whitespaces https://stackoverflow.com/a/19661428/810109?

Reply:
Oh I just did it that way because I saw `sed` was used elsewhere.  Yes it looks like you can do this in plain bash.  I wasn't able to get the above example working (it seems to replace all spaces, not just beginning and ending), but this approach seems to work:  
```
str="  #  1.2.3  "
modStr="${str#"${str%%[![:space:]]*}"}"
modStr="${modStr%"${modStr##*[![:space:]]}"}"
echo "modStr is [$modStr]"
# prints '# 1.2.3'
```
That will work in POSIX compatible bashes.

Reply:
Ref: answer by Mateusz Piotrowski at [SOF](https://stackoverflow.com/questions/369758/how-to-trim-whitespace-from-a-bash-variable)

Reply:
I pushed a new version using a function derived from the above.